src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -139,12 +155,17 @@ public Workspace Workspace
     /// </summary>
     public Project? GetProject(ProjectId? projectId)
     {
-        if (this.ContainsProject(projectId))
+        if (projectId is null || !_projectIdToIndex.TryGetValue(projectId, out var projectIndex))
+            return null;
+
+        var project = _projects[projectIndex];
+        if (project == null)
         {
-            return ImmutableInterlocked.GetOrAdd(ref _projectIdToProjectMap, projectId, s_createProjectFunction, this);
+            project = CreateProject(projectId, this);
```

Comment:
can you doc that this is fine to invoke, and hten throw away (it def has to be, as that was the semantics from before), but it would be good to state it.